### PR TITLE
feat: Add playground to website

### PR DESCRIPTION
**Pull request by phated**
It's time for @spotandjake's wildest dreams to come true.

Casually ignore the committed 12mb grainc.bc.mjs - don't have a good way to automatically pull that into the repo.

### CODE REVIEW

**Comment by marcusroberts** [comment]
When I try the preview build, it compiles and runs fine, but any changes I make to the source are then ignored in future compiles

**Comment by phated** [comment]
> When I try the preview build, it compiles and runs fine, but any changes I make to the source are then ignored in future compiles

This is not enough information to explain what you are doing (or doing wrong). I can say for a fact that I developed everything with `npm run start-playground` and the rebuilds work perfectly.

**Comment by marcusroberts** [comment]
When I click the Deploy Preview link to Netlify in his PR. (https://deploy-preview-316--grain-lang.netlify.app/playground) and press Compile and Run, it works fine, but any changes I make in the source code panel aren't reflected when I press Compile and Run a second time.

**Comment by phated** [comment]
> When I click the Deploy Preview link to Netlify in this PR. (https://deploy-preview-316--grain-lang.netlify.app/playground) and press Compile and Run, it works fine, but any changes I make in the source code panel aren't reflected when I press Compile and Run a second time.

There is a bug in the js-runner that @ospencer needs to look into that causes it to be stateful and break multiple runs.

**Comment by phated** [comment]
@marcusroberts I dropped Grain's js-runner completely and switched to using the new 1.x version of `@wasmer/wasi` directly.

I tried a few simple examples but can you test your stuff again once the new deploy finishes?

**Comment by marcusroberts** [comment]
That fixed it!

**Review by marcusroberts — APPROVED** [review]
Works great for me!

**Review by ospencer — APPROVED** [review]
Excellent excellent work!

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,5 @@ _multiconfig.yml
 /docs
 /guide
 /blog
+/playground
```
Comment (phated) [inline .gitignore:11]:
```suggestion
/try
```

File: vite.config.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const path = require("path");
+
+const root = path.join(__dirname, "./playground_src");
+
+module.exports = ({ mode }) => {
+  return {
+    root,
+    base: mode === "development" ? "/" : "/playground/",
```
Comment (phated) [inline vite.config.js:8]:
```suggestion
    base: mode === "development" ? "/" : "/try/",
```

File: vite.config.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          process: JSON.stringify({}),
+          "process.env": JSON.stringify({}),
+          "process.env.NODE_DEBUG": "false",
+        },
+      },
+    },
+
+    build: {
+      target: "es2020",
+      outDir: "../playground",
```
Comment (phated) [inline vite.config.js:39]:
```suggestion
      outDir: "../try",
```

**Review by phated — COMMENTED** [review]
Talked with @ospencer and I think we want this URL to be `/try`. These changes should do that

File: index.html [hunk 4]
```diff
@@ -52,6 +52,7 @@
         <a href="/docs/intro" class="nav-link heading">Documentation</a>
         <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
         <a href="/blog" class="nav-link heading">Blog</a>
+        <a href="/playground" class="nav-link heading">Try</a>
```
Comment (phated) [inline index.html:55]:
```suggestion
        <a href="/try" class="nav-link heading">Try</a>
```

File: index.html [hunk 5]
```diff
@@ -64,6 +65,7 @@
       <a href="/docs/intro" class="nav-link heading">Documentation</a>
       <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
       <a href="/blog" class="nav-link heading">Blog</a>
+      <a href="/playground" class="nav-link heading">Try</a>
```
Comment (phated) [inline index.html:68]:
```suggestion
      <a href="/try" class="nav-link heading">Try</a>
```

File: playground_src/index.html [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+<body>
+    <section id="hero">
+        <nav id="nav">
+            <a href="/"><img class="nav-logo" width="127" height="32" src="/images/grain_logo_white.svg"></a>
+            <div class="nav-links">
+                <a href="/docs" class="nav-link heading">Guide</a>
+                <a href="/docs/intro" class="nav-link heading">Documentation</a>
+                <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
+                <a href="/blog" class="nav-link heading">Blog</a>
+                <a href="/playground" class="nav-link heading">Try</a>
```
Comment (phated) [inline playground_src/index.html:63]:
```suggestion
                <a href="/try" class="nav-link heading">Try</a>
```

File: playground_src/index.html [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+                <a href="https://discord.com/invite/grain-lang" class="nav-link"><i
+                        class="fab fa-discord nav-social-link"></i></a>
+            </div>
+        </nav>
+        <div class="nav-links-mobile">
+            <a href="/docs" class="nav-link heading">Guide</a>
+            <a href="/docs/intro" class="nav-link heading">Documentation</a>
+            <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
+            <a href="/blog" class="nav-link heading">Blog</a>
+            <a href="/playground" class="nav-link heading">Try</a>
```
Comment (phated) [inline playground_src/index.html:79]:
```suggestion
            <a href="/try" class="nav-link heading">Try</a>
```

File: themes/grain/layout/_partial/hero.ejs [hunk 8]
```diff
@@ -13,6 +13,7 @@
       <a href="/docs/intro" class="nav-link heading">Documentation</a>
       <a href="https://github.com/grain-lang/grain/graphs/contributors" class="nav-link heading">Community</a>
       <a href="/blog" class="nav-link heading">Blog</a>
+      <a href="/playground" class="nav-link heading">Try</a>
```
Comment (phated) [inline themes/grain/layout/_partial/hero.ejs:16]:
```suggestion
      <a href="/try" class="nav-link heading">Try</a>
```

**Pull request merged**